Drivers/STM32F4xx_HAL_Driver/Inc/stm32f4xx_hal.h

```diff
@@ -201,7 +201,7 @@ typedef enum
 /** @addtogroup HAL_Exported_Variables
   * @{
   */
-extern __IO uint32_t uwTick;
+extern volatile uint32_t uwTick;

```

Comment:
Nie powinieneś zmieniać ręcznie nic w bibliotekach hala. Chyba, że to cube nadpisał?